extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/VertxHttpHotReplacementSetup.java

```diff
@@ -119,6 +119,7 @@ public void handle(Promise<Object> promise) {
                             } catch (Throwable t) {
                                 routingContext.fail(t);
```

Comment:
@cescoffier  actually I think that the routing context should be failed in the future returned by executeBlocking instead of the block

Reply:
Agreed, using `.onFailure(routingContext::fail);` makes more sense.